wordpress-mobile/WordPress-iOS
# Log events for domain credit analytics

Refs. #11467 

* New analytics event enums were added in https://github.com/wordpress-mobile/WordPress-iOS-Shared/pull/204
* Specified event names for new enums in `WPAnalyticsTrackerAutomatticTracks`.
* Logged events in 3 cases:
  * `WPAnalyticsStatDomainCreditPromptShown`: logged on the first time when domain credit UI is shown during site dashboard lifetime. In `BlogDetailsViewController`, table view data could be reloaded more than once like in every `viewWillAppear`. Added a local variable to track whether the event has been logged.
  * `WPAnalyticsStatDomainCreditRedemptionTapped`: when user taps on domain credit redemption call to action ("Register Domain" row).
  * `WPAnalyticsStatDomainCreditRedemptionSuccess`: logged on domain registration success and when user sees congratulations screen.

Blocking tasks:
- [ ] merge https://github.com/wordpress-mobile/WordPress-iOS-Shared/pull/204
- [ ] update `Podfile` for new `WordPress-iOS-Shared` version `1.8.0-beta.1`

To test:

- Prerequisites: have a site/blog with paid WordPress.com plan (e.g. Blogger plan) and its domain credit still available
- Go to a site dashboard where the site meets conditions in prerequisites, domain credit section should be shown --> should see `🔵 Tracked: domain_credit_prompt_shown` in console
- Tap on "Register Domain" row --> should see `🔵 Tracked: domain_credit_redemption_tapped` in console
- Pick a `*.blog` domain (`.blog` has high purchase volume so that cancellation rate from testing has less effect on production domain service)
- Fill out registration details (these will be saved for the next time)
- Submit domain registration, and wait for registration service to complete
- Around the time an email on domain registration success is received, the domain registration modal should be dismissed and a congratulations screen is shown with the purchased domain --> should see `🔵 Tracked: domain_credit_redemption_success` in console
- After dismissing the congratulations screen, a snackbar should appear notifying further instructions in email. At the same time, domain credit UI should not be shown on site dashboard
- **[Important]** Cancel domain purchase shortly after testing in Account Profile --> Manage Purchases ([official FAQ](https://en.support.wordpress.com/manage-purchases/#FAQ-Cancelling))

Update release notes:
